manifests/unit.pp

```diff
@@ -22,55 +22,69 @@
   $install_wantedby        = undef,
 ) {
 
-  validate_re($ensure, [ '^present$', '^absent$' ],
-    "systemd::unit::${name}::ensure is invalid and does not match the regex.")
+  include ::systemd
+
+  if is_string($ensure) {
+    validate_re($ensure, [ '^present$', '^absent$' ],
+      "systemd::unit::${name}::ensure is not a string or is invalid and does not match the regex.")
+  } else {
+    fail("systemd::unit::${name}::ensure is not a string or is invalid and does not match the regex.")
+  }
 
   validate_absolute_path($systemd_path)
 
-  if $unit_after != undef {
-    validate_string($unit_after)
+  if $unit_after != undef and is_string($unit_after) == false {
```

Comment:
why replacing a deprecated function with another one?

Reply:
This is also, as before, to ensure the module doesn't stop working when non-strings are sent as values to the parameter.

 Note: This function is compatible only with Puppet 4.4.0 (PE 2016.1) and later.